Library/Homebrew/download_strategy.rb

```diff
@@ -543,7 +550,7 @@ def _curl_args
 
     args += ["--user", meta.fetch(:user)] if meta.key?(:user)
 
-    args += [meta[:header], meta[:headers]].flatten.compact.flat_map { |h| ["--header", h.strip] }
+    args += [meta[:headers]].flatten.compact.flat_map { |h| ["--header", h.strip] }
```

Comment:
```suggestion
    args += meta.fetch(:headers, []).map { |h| ["--header", h.strip] }
```

Reply:
Hi @aacerox, @reitermarkus: this change seems to have some unintended change in behavior that has broken one of our private taps where we set up headers like this:
```
        meta[:headers] ||= []
        meta[:headers] << ["Accept: application/octet-stream"]
```
And now it fails like this:
```
Error: undefined method `strip' for ["Accept: application/octet-stream"]:Array
Please report this issue:
  https://docs.brew.sh/Troubleshooting
/opt/homebrew/Library/Homebrew/download_strategy.rb:564:in `block in _curl_args'
```

This works with the previous version of brew. I think this is because the new code no longer calls `flatten` on the headers so it tries to call `strip` on our header while it is an array instead of flattening it to a string. Do you think we can add back the `flatten` to maintain compatibility?

Reply:
Yeah, I realize I am like the [spacebar guy from xkcd](https://xkcd.com/1172/) right now. I don't mind making that change on my side, but I wanted to bring this up since it is probably an unintended side effect of this change that may affect others as well.